### PR TITLE
ROX-17848: Show report review details

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

This PR lists all the selected report parameters, the delivery destinations, and schedule details in the wizard's `Review and create` section. For now, I filled in the delivery destinations and schedule details with fake data. I can also leave it blank until I work on those sections. We also display the CVE report format at the end. This is just a sample of what it looks like—not actually using the data. 

Screenshot 2023-07-10 at 9 53 04 AM
Screenshot 2023-07-10 at 9 53 05 AM

### CODE REVIEW

**Comment by sachaudh** [comment]
@zhenpesky please take a look at the screenshot and let me know if it looks good or needs any adjustment

File: ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/components/VulnerabilitySeverityIconText.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/components/VulnerabilitySeverityIconText.tsx:1]: Could you use https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Components/PatternFly/IconText/VulnerabilitySeverityIconText.tsx the same way you are using this new component?

File: ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/utils.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    IMPORTANT_VULNERABILITY_SEVERITY: 'Important',
+    MODERATE_VULNERABILITY_SEVERITY: 'Moderate',
+    LOW_VULNERABILITY_SEVERITY: 'Low',
+};
+
+export const cveStatusLabelMap: Record<CVEStatus, string> = {
+    FIXABLE: 'Fixable',
+    NOT_FIXABLE: 'Not fixable',
+};
+
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/utils.ts:17]: Similarly with these, could you use https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/constants/reportConstants.ts instead?

Although severity and fixability are general enough that an argument can be made to put them in a file with a name less specific than `reportConstants`...

**Review by zhenpesky — APPROVED** [review]

File: ui/apps/platform/src/hooks/usePaginatedQuery.ts [hunk 2]
```diff
@@ -84,8 +84,8 @@ export function usePaginatedQuery<Item, ItemKey>(
     // the `fetchPageHandler` function below relies on the reference of `dedupKeyFn`. By storing these
     // in state, it removes the burden of wrapping in a useCallback or useMemo from the caller.
     const [{ debounceRate, dedupKeyFn, onError, manualFetch = false }] = useState(initialOptions);
-    const [isRefreshingResults, setIsRefreshingResults] = useState(false);
-    const [isFetchingNextPage, setIsFetchingNextPage] = useState(false);
+    const [isRefreshingResults, setIsRefreshingResults] = useState(true);
+    const [isFetchingNextPage, setIsFetchingNextPage] = useState(true);
```
Comment (dvail) [inline ui/apps/platform/src/hooks/usePaginatedQuery.ts:88]: This makes sense as a default, but I think the semantics will be weird if `initialOptions.manualFetch` is set to `true`. In that case, the hook will declare that a fetch is in progress before one ever is initialized. Can we default this to `true` in all cases except for when `initialOptions.manualFetch` is set to `true`?

File: ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/ReportReviewForm.tsx [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+                className="pf-u-py-lg pf-u-px-lg"
+            >
+                <Flex direction={{ default: 'column' }}>
+                    <FlexItem>
+                        <Title headingLevel="h3">Report parameters</Title>
+                    </FlexItem>
+                    <FlexItem flex={{ default: 'flexNone' }}>
+                        <DescriptionList
+                            columnModifier={{
+                                default: '2Col',
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/ReportReviewForm.tsx:82]: Nit for all those users running ACS on their phone 😅: Could we do this here and in the section below?
```suggestion
                                default: '1Col',
                                sm: '2Col',
```

File: ui/apps/platform/src/hooks/usePaginatedQuery.ts (hunk 2 above)
Comment (dvail) [inline ui/apps/platform/src/hooks/usePaginatedQuery.ts:88]:
```suggestion
    const [isRefreshingResults, setIsRefreshingResults] = useState(!manualFetch);
    const [isFetchingNextPage, setIsFetchingNextPage] = useState(!manualFetch);
```

**Review by dvail — APPROVED** [review]
Just a few small comments but overall this was an easy read 👍

File: ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/ReportReviewForm.tsx (hunk 3 above)
Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/ReportReviewForm.tsx:82]: Did you mean default is `2Col` and sm is `1Col`? There was a weird issue where it wouldn't go back to `2Col` when I stretched it out again. I added and md as well and seems to work now

**Pull request merged**